### PR TITLE
Docs for Cubical indexed types

### LINKED ISSUE #6227: CHANGELOG: Document handling of indexed data types in Cubical, plus the warning `NoEquivWhenSplitting`
E.g. the test case for #4725 generates this warning:
```
Could not generate equivalence when splitting on indexed family,
the function will not compute on transports by a path.
  Reason: UnsupportedYet Injectivity
                           position:    0
                           type:        D
                           datatype:    D
                           parameters: 
                           indices:    
                           constructor: d
when checking the definition of g
```
I dimly remember we discussed adding a flag to switch off these warnings. Shouldn't we provide such a flag with 2.6.3?
@plt-amy @jespercockx @nad @Saizan

### PR DESCRIPTION
**Pull request by plt-amy**
PR just because I've had bad times with reST before, and it would be embarrassing to mess that up on master. (Closes #6227)

### CODE REVIEW

**Comment by andreasabel** [comment]
Rendering here: https://agda--6232.org.readthedocs.build/en/6232/language/cubical.html

File: doc/user-manual/language/cubical.lagda.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ..
   ::
 
-  {-# OPTIONS --cubical #-}
+  {-# OPTIONS --cubical -vtc.lhs.unify.inv:20 -WNoEquivWhenSplitting #-}
```
Comment (andreasabel) [inline doc/user-manual/language/cubical.lagda.rst]: Do these extra options have any effect on the user-manual or its build?

Comment (andreasabel) [reply doc/user-manual/language/cubical.lagda.rst]: I removed them in my fixup, but feel free to restore them if that wasn't good.

**Review by andreasabel — COMMENTED** [review]
I committed some fixes directly.

**Comment by andreasabel** [comment]
Er, I did the edit directly on github, and I think it screwed up when your commit came inbetween.

Let me repair this...

File: doc/user-manual/language/cubical.lagda.rst (hunk 1 above)
Comment (plt-amy) [reply doc/user-manual/language/cubical.lagda.rst]: Ah, thanks; they were just to help me figure out a minimal reproducer for the Setω transport case. (I think I'm going to write a pre-commit hook that looks through the diff and asks if I really meant to commit the verbosity flags....)

**Comment by andreasabel** [comment]
Ok, fixed my commit.

**Review by andreasabel — APPROVED** [review]
From my side, this looks good, modulo I do not really understand Cubical enough to fully follow.  Maybe @mortberg wants to review...

**Comment by andreasabel** [comment]
Looks like we rather need option `-WnoNoEquivWhenSplitting` (mmh, doubly-negated warning):
```
UserManual
    language-cubical:                              FAIL (0.89s)
                                                   Result did not match golden value.
                                                   Diff between actual and golden value:
...
+Could not generate equivalence when splitting on indexed family,
+the function will not compute on transports by a path.
+  Reason: UnsupportedYet Injectivity
+                           position:    0
+                           type:        Nat
+                           datatype:    Nat
+                           parameters: 
+                           indices:    
+                           constructor: suc
+when checking the definition of sucInjEq
```

**Comment by andreasabel** [comment]
Let me fix the conflict I caused.

**Comment by andreasabel** [comment]
I merged in `master`, we need to __SQUASH__ when merging this PR.

File: doc/user-manual/language/cubical.lagda.rst [hunk 2]
```diff
@@ -726,8 +729,143 @@ is defined as:
...
+
+  module _ {a} {A B : Set a} {x y : A} {f : A ≃ B} where
+    _ : symEq (reflEq {x = x}) ≡ reflEq
+    _ = refl
+
+    _ : transpEq (pathToEq (ua (idEquiv Bool))) ≡ λ x → x
+    _ = refl
+
+The computational behaviour of matching on indexed inductives in very
+generic is suboptimal when compared to the behaviour of similar
```
Comment (andreasabel) [inline doc/user-manual/language/cubical.lagda.rst]: I don't see the transition of thought from "suboptimal" (negative) to the example that follows, where "we can investigate" (positive).  There is a gap in the argument, imo.

Can you reformulate to clarify?

File: doc/user-manual/language/cubical.lagda.rst [hunk 3]
```diff
@@ -726,8 +729,143 @@ is defined as:
...
+      compPath (transportRefl (f .fst _))
+        (cong (f .fst) (compPath
+          (transportRefl _)
+          (compPath
+            (transportRefl _)
+            (transportRefl _))))
+
+In more concrete situations, such as when the indices are constructors
+of some other inductive type, pattern-matching definitions will not
+compute when applied to transports.
```
Comment (andreasabel) [inline doc/user-manual/language/cubical.lagda.rst]: Would this warrant an example?

Comment (andreasabel) [reply doc/user-manual/language/cubical.lagda.rst]: Ah I see this comes later, but is interrupted by the `NoEquivWhenSplitting` discourse.

Please add the usual forward referencing.

File: doc/user-manual/language/cubical.lagda.rst [hunk 4]
```diff
@@ -726,8 +729,143 @@ is defined as:
...
+``agda-lib`` file.
+
+What works, and what doesn't
+----------------------------
+
+This section lists some of the common cases where pattern-matching
+unification produces something that can not be extended to cover
+transports, and the cases in which it can. The following pair of
+definitions relies on injectivity for data constructors, and so will not
+compute on non-``reflEq`` paths:
```
Comment (andreasabel) [inline doc/user-manual/language/cubical.lagda.rst]: Where will this hit me?  E.g. what lemma on `sucInjEq` will then fail to go through?

File: doc/user-manual/language/cubical.lagda.rst [hunk 5]
```diff
@@ -726,8 +729,143 @@ is defined as:
...
+  sucInjEq : ∀ {n k} → Eq (suc n) (suc k) → Eq n k
+  sucInjEq reflEq = reflEq
+
+  trueDel : Eq true true → Eq true true
+  trueDel reflEq = reflEq
+
+In some cases, it may be possible to rephrase the proof in ways that
+avoid pattern matching, and so, compute. As a simple example, since
+``apEq`` below will always compute, we can use the fact that ``suc`` has
+a partially-defined inverse to conclude it is injective:
```
Comment (andreasabel) [inline doc/user-manual/language/cubical.lagda.rst]: I remember this trick from my apprentice years in type theory.

This seems to document a workaround.  Ok.  Might a good advice to users...

**Review by andreasabel — CHANGES_REQUESTED** [review]
I sat down again in a quieter moment and can say that I mostly follow this exposition.
I added some comments, some with a request to clarify the formulation.
Thanks for the docs!

**Review by andreasabel — APPROVED** [review]
Great job, thanks!

**Pull request merged**